### PR TITLE
Add HAMRONIZATION/FARGENE and updates all HARMONIZATION tools to 1.1.1

### LINKED ISSUE #2106: new module: HAMRONIZATION/FARGENE
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #2106 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: modules/hamronization/fargene/meta.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: "hamronization_fargene"
+description: Tool to convert and summarize fARGene outputs using the hAMRonization specification.
```
Comment (jasmezz) [inline modules/hamronization/fargene/meta.yml:2]:
```suggestion
description: Tool to convert and summarize fARGene outputs using the hAMRonization specification
```

File: modules/hamronization/fargene/meta.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - report:
+      type: file
+      description: Output .txt file from RGI
```
Comment (jasmezz) [inline modules/hamronization/fargene/meta.yml:27]:
```suggestion
      description: Output .txt file from fARGene
```

File: modules/hamronization/fargene/meta.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      pattern: "*.txt"
+  - format:
+      type: value
+      description: Type of report file to be produced
+      pattern: "tsv|json"
+  - software_version:
+      type: value
+      description: Version of DeepARG used
```
Comment (jasmezz) [inline modules/hamronization/fargene/meta.yml:35]:
```suggestion
      description: Version of fARGene used
```

File: modules/hamronization/fargene/meta.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  - amr
+  - antimicrobial resistance
+  - arg
+  - antimicrobial resistance genes
+  - reporting
+  - fARGene
+tools:
+  - hamronization:
+      description: "Tool to convert and summarize AMR gene detection outputs using the hAMRonization specification"
+      homepage: "https://github.com/pha4ge/hAMRonization/blob/master/README.md"
```
Comment (jasmezz) [inline modules/hamronization/fargene/meta.yml:13]: I just noticed (also for other modules) that this points to the `README.md`s. Is that preferred or better point to the whole repo?

File: modules/hamronization/fargene/meta.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      type: value
+      description: Type of report file to be produced
+      pattern: "tsv|json"
+  - software_version:
+      type: value
+      description: Version of DeepARG used
+      pattern: "[0-9].[0-9].[0-9]"
+  - reference_db_version:
+      type: value
+      description: Database version of DeepARG used
```
Comment (jasmezz) [inline modules/hamronization/fargene/meta.yml:39]:
```suggestion
      description: Database version of fARGene used
```

**Review by jasmezz — CHANGES_REQUESTED** [review]
Just some typos. Since you are still working on the tests to pass, I might have a look later again.

File: modules/hamronization/fargene/meta.yml (hunk 4 above)
Comment (jfy133) [reply modules/hamronization/fargene/meta.yml]: Ah just the repo... will update that!

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Review by jasmezz — APPROVED** [review]
Looking good now!

**Comment by jfy133** [comment]
Conda is flaking out, but containers are working, so going to merge now :)

**Pull request merged**